src/ucp/core/ucp_am.c

```diff
@@ -966,17 +969,21 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
                                  goto out;});
 
     ucp_am_send_req_init(req, ep, header, header_length, buffer, datatype,
-                         count, flags, id);
+                         count, flags, id, ucp_request_param_mem_type(param));
 
     if (flags & UCP_AM_SEND_REPLY) {
+        /* Note that max_eager_short.memtype_on is always initialized to real
+         * max_short value
+         */
         ret = ucp_am_send_req(req, count, &ucp_ep_config(ep)->am, param,
-                              ucp_ep_config(ep)->am_u.reply_proto,
-                              ucs_min(ucp_ep_config(ep)->am_u.max_eager_short,
-                                      UCP_AM_SHORT_REPLY_MAX_SIZE), flags);
+                          ucp_ep_config(ep)->am_u.reply_proto,
+                          ucs_min(ucp_ep_config(ep)->am_u.max_eager_short.memtype_on,
+                          UCP_AM_SHORT_REPLY_MAX_SIZE), flags);
```

Comment:
should `UCP_AM_SHORT_REPLY_MAX_SIZE` be aligned as:
```
...
ucp_ep_config(ep)->am_u.reply_proto,
ucs_min(ucp_ep_config(ep)->am_u.max_eager_short.memtype_on,
        UCP_AM_SHORT_REPLY_MAX_SIZE),
flags);
```

also, can save config to separate variable, so it it will be shorten